### PR TITLE
allow changing textAlign of TextField in DropdownMenu

### PR DESCRIPTION
**Pull request by angelocordero**
PR for #147991 

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[Data Driven Fixes]: https://github.com/flutter/flutter/wiki/Data-driven-Fixes

### CODE REVIEW

**Review by ghost — COMMENTED** [review]
ok

File: packages/flutter/lib/src/material/dropdown_menu.dart [hunk 1]
```diff
@@ -154,6 +154,7 @@ class DropdownMenu<T> extends StatefulWidget {
     this.enableFilter = false,
     this.enableSearch = true,
     this.textStyle,
+    this.textAlign,
```
Comment (justinmc) [inline packages/flutter/lib/src/material/dropdown_menu.dart:157]: I think you should put the default value (`TextAlign.start`) here explicitly.

**Review by justinmc — COMMENTED** [review]
@angelocordero Can you add a test? I think just something that verifies that it was passed through is probably fine if that's easiest.

**Comment by angelocordero** [comment]
@justinmc done adding a test and fixed the default value

**Review by justinmc — APPROVED** [review]
LGTM :+1:

File: packages/flutter/lib/src/material/dropdown_menu.dart [hunk 2]
```diff
@@ -264,6 +265,11 @@ class DropdownMenu<T> extends StatefulWidget {
   /// if the dropdown menu theme's value is null.
   final TextStyle? textStyle;
 
+  /// The text align for the [TextField] of the [DropdownMenu];
```
Comment (Piinks) [inline packages/flutter/lib/src/material/dropdown_menu.dart:268]:
```suggestion
  /// The text align for the [TextField] of the [DropdownMenu].
```

File: packages/flutter/lib/src/material/dropdown_menu.dart [hunk 3]
```diff
@@ -264,6 +265,11 @@ class DropdownMenu<T> extends StatefulWidget {
...
+  ///
+  /// Defaults to [TextAlign.start]
```
Comment (Piinks) [inline packages/flutter/lib/src/material/dropdown_menu.dart:270]:
```suggestion
  /// Defaults to [TextAlign.start].
```

**Review by Piinks — APPROVED** [review]
Welcome @angelocordero thanks for contributing! LGTM with a few nits

**Comment by justinmc** [comment]
@angelocordero Can you push a merge commit? Not sure why the ci.yaml validation is failing.

**Comment by Piinks** [comment]
The Google testing failure here was an infrastructure issue, the tests were not able to run. I have rebased this to give the bots a kick, hopefully everything will pass this run. :)

**Pull request merged**